src/macaron/malware_analyzer/pypi_heuristics/metadata/wheel_absence.py

```diff
@@ -64,21 +69,46 @@ def analyze(self, pypi_package_json: PyPIPackageJsonAsset) -> tuple[HeuristicRes
                 logger.debug(error_msg)
                 raise HeuristicAnalyzerValueError(error_msg)
 
-        release_files: list[JsonType] = []
+        inspector_links: list[JsonType] = []
         wheel_present: bool = False
 
         try:
             for release_metadata in releases[version]:
                 if release_metadata["packagetype"] == self.WHEEL:
                     wheel_present = True
 
-                release_files.append(release_metadata["filename"])
+                name = json_extract(pypi_package_json.package_json, ["info", "name"], str)
+                if name is None:
+                    error_msg = "There is no 'name' field for this package."
+                    logger.debug(error_msg)
+                    raise HeuristicAnalyzerValueError(error_msg)
+
+                blake2b_256 = release_metadata["digests"]["blake2b_256"]
```

Comment:
Should we use the `json_extract` method for extracting this value and validate it's a string that has more than 4 characters ?

Reply:
This information will exist and be generated by pypi if we get to this line, so a scenario where the hash doesn't exist or is malformed should never happen if the release information is actually there (which is checked earlier). If for whatever reason this does crash, it would cause an exception and the result to be UNKNOWN, which is what we would throw for a custom exception anyway. I think this is safe to leave for now. 

`json_extract` doesn't really need to be used on this since it would need to start from the top (i.e. `["releases"]["version"]["digests"]["blake2b_256"]).

Reply:
I've refactored it to use `json_extract` on all accesses and raise exceptions now, making the code a little cleaner and addressing your issue as well.